doc/community/developer-guide.md

```diff
@@ -135,3 +153,51 @@ with pytest.raises(ZeroDivisionError) as excinfo:
 ```{code-cell} ipython3
 assert str(excinfo.value) == "division by zero"
 ```
+
+## Integration tests
+
+Integration tests check compatibility with different databases. They are executed on
+each PR; however, you might need to run them locally.
+
+```{note}
+Setting up the development environment for running integration tests locally
```

Comment:
Perhaps, consider adding another note here to push developers to rely on the CI to test PRs since integration tests are not yet supported on all OSes

Reply:
It might also be helpful to include which OSes are supported